WISE-Community/WISE
# Make WISE run with Spring-Boot

WISE is now a spring-boot application, which means that many of the dependencies and configurations are done for us, including embedded tomcat server and monitoring. Many dependencies were thus removed from pom.xml. We also removed web.xml.

To run WISE under Spring-Boot, you need to run the main method in Application.java. This is different from how we've been running WISE until now (mvn tomcat7:run). If you're using the IDE, you should edit your run configurations. If you've been running with wise.sh, it's still the same: ./wise.sh dev.

How to test
0. Run mvn test. Make sure that all tests pass.
1. Run Application.java in your IDE. Make sure that everything works as before. The new site is now available under "/" instead of "/wise".
2. Run WISE via command line: "./wise.sh dev". This now calls "mvn spring-boot:run". If you verified that the site works in step 1, you just need to do a cursory test in this one.
3. Make a war file with "./wise.sh package". Verify that target/wise.war is created.
